src/main/java/teammates/ui/webapi/action/DeleteInstructorInCourseAction.java

```diff
@@ -71,4 +76,27 @@ private boolean hasAlternativeInstructor(String courseId, String instructorToDel
         return false;
     }
 
+    /**
+     * Returns true if there is a joined instructor that is visible to students(other than the instructor to delete)
+     * with the privilege of modifying instructors.
+     *
+     * @param courseId                Id of the course
+     * @param instructorToDeleteEmail Email of the instructor who is being deleted
+     */
+    private boolean hasAlternativeInstructorDisplayed(String courseId, String instructorToDeleteEmail) {
```

Comment:
You should combine this with the previous method `hasAlternativeInstructor`. Otherwise you're duplicating the DB call in `logic.getInstructorsForCourse`.

So that method needs to check that there is at least an instructor:
- Who is visible
- Who can modify instructors

And it may or may not be the same instructor.

Reply:
in essence, just add a isDisplayedtoStudents() check in the previoud methods along with the other checks?